### PR TITLE
Adding high level date and datetime example to base structs, fixing typo

### PR DESCRIPTION
**Pull request by andrewpollack**
* Copying ISO examples of `Date` and `DateTime` to base structs, increasing immediate accessibility to structs
* Fixing a `date` --> `datetime` typo

### CODE REVIEW

File: components/calendar/src/date.rs [hunk 1]
```diff
@@ -34,10 +34,21 @@ impl<C: Calendar> AsCalendar for Rc<C> {
...
 ///
 /// This can work with wrappers around [`Calendar`] types,
-/// e.g. `Rc<C>`, via the [`AsCalendar`] trait
+/// e.g. `Rc<C>`, via the [`AsCalendar`] trait.
+///
+/// ```rust
+/// use icu::calendar::Date;
+///
+/// // Example: creation of ISO date from integers.
+/// let date_iso = Date::new_iso_date_from_integers(1970, 1, 2).unwrap();
```
Comment (zbraniecki) [inline components/calendar/src/date.rs:46]: `nit`: please, use `expect` over `unwrap` - having `expect` everywhere in visible code/docs helps to understand what a failure means.

**Review by zbraniecki — DISMISSED** [review]

File: components/calendar/src/date.rs (hunk 1 above)
Comment (andrewpollack) [reply components/calendar/src/date.rs:46]: Thank you for the feedback! For my understanding of Rust, are there cases where `unwrap` would be preferred?

Comment (zbraniecki) [reply components/calendar/src/date.rs]: You can use it in integration tests and unit tests which are not exposed to readers. When i code I often use unwrap for convenience and then swap for expect when i clean up for review. It also gives me a chance to review if i can bubble up, and if a given fallibility is still needed (sometimes you'll add it at some point and then refactor removing the reason but keep the signature. It's a good chance to revisit)

**Review by zbraniecki — APPROVED** [review]

File: components/calendar/src/date.rs (hunk 1 above)
Comment (andrewpollack) [reply components/calendar/src/date.rs]: That makes sense -- thanks for the explanation!

**Review by Manishearth — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
lgtm with follow-up in #1922

**Pull request merged**